### PR TITLE
ENH: Move database error to error/__init__.py per GH27656

### PR DESCRIPTION
**Pull request by dataxerik**
- [x] xref #27656. this GitHub issue is being done in multiple parts
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/errors/__init__.py [hunk 1]
```diff
@@ -456,12 +456,30 @@ class AttributeConflictWarning(Warning):
     """
 
 
+class DatabaseError(OSError):
+    """
+    Error is raised when executing sql with bad syntax or sql that throws an error.
+
+    Examples
+    --------
+    >>> import pandas.io.sql as sql
```
Comment (mroeschke) [inline pandas/errors/__init__.py:465]: Can just call `pd.read_sql` below

Comment (mroeschke) [reply pandas/errors/__init__.py:465]: I don't think it's necessary to demo `sql.execute` since it's not public

Comment (dataxerik) [reply pandas/errors/__init__.py]: Updated. Maybe an obvious question, how do you know if something is public or not?

Comment (mroeschke) [reply pandas/errors/__init__.py]: Good question (and definitely not obvious). Anything in the pandas namespace `import pandas as pd` or `pandas/api` is public. Users can access other namespaces but they are not "public" e.g. `pandas.core`

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @dataxerik

File: pandas/errors/__init__.py (hunk 1 above)
Comment (dataxerik) [reply pandas/errors/__init__.py]: Gotcha, thank you for the information!

**Pull request merged**